plugins/ingest/src/main/java/org/elasticsearch/ingest/processor/simple/SimpleProcessor.java

```diff
@@ -55,10 +55,10 @@ public void execute(Data data) {
     public static class Factory implements Processor.Factory {
 
         public Processor create(Map<String, Object> config) {
```

Comment:
I guess it is a design decision for future integrations to have a Map for the config instead of a proper java object? If we had the latter (e.g. a Configuration object) the readStringProperty that is now in ConfigurationUtils could be part of that object. Seems a bit cleaner...not a big deal though